Extensions/mutualchecker.js

```diff
@@ -0,0 +1,83 @@
+//* TITLE Mutual Checker **//
+//* VERSION 1.0.0 **//
+//* DESCRIPTION Adds an icon next to usernames to indicate if they follow you. **//
+//* DETAILS Does what it says on the tin! Doesn't work on sideblogs. Made from fresh, distilled Profiler juice. **//
+//* DEVELOPER AprilSylph **//
+//* FRAME false **//
+//* BETA false **//
+
+XKit.extensions.mutualchecker = new Object({
+
+	running: false,
+
+	add_icons: function() {
+		if ($(".follower").length > 0) {
+			$(".follower").each(function() {
+				var m_url = $(this).find(".name").text();
+				$(this).find(".name").append("<span class=\"mutualcheck " + m_url + "\"></span>");
+			});
+		}
+		var posts = XKit.interface.get_posts("mutualchecker-done");
+		$(posts).each(function() {
+			$(this).addClass("mutualchecker-done");
+			var m_post = XKit.interface.post($(this));
+			if (XKit.interface.where().inbox !== true) {
+				if (m_post.is_mine === true) { return; }
+			}
+			var post_owner = m_post.owner;
+			if (m_post.type === "note" && XKit.interface.where().inbox === true) {
+				var m_json_info = $(this).find(".post_avatar_link").attr('data-tumblelog-popover');
+				try {
+					var m_json_obj = JSON.parse(m_json_info);
+					post_owner = m_json_obj.name;
+				} catch (e) {
+					return;
+				}
+			}
+			var name_div_container;
+			var name_div;
+			if ($(this).find(".post_info_fence").length > 0) {
+				name_div_container = $(this).find(".post_info_fence");
+			} else {
+				name_div_container = $(this).find(".post_info");
+			}
+			if (name_div_container.find("a").length > 0) {
+				name_div = name_div_container.find("a").first();
+				$(name_div).after("<span class=\"mutualcheck " + post_owner + "\"></span>");
+			} else {
+				name_div = name_div_container;
+				$(name_div).append("<span class=\"mutualcheck " + post_owner + "\"></span>");
+			}
+			var blog_id;
+			var m_blogs = XKit.tools.get_blogs();
+			for (var i = 0; i < m_blogs.length; i++) {
+				if (m_blogs[i] !== "") {
+					blog_id = m_blogs[i];
+					break;
+				}
+			}
```

Comment:
what is going on here. is there a problem with just using get_blogs()[0]? I tested get_blogs on a couple of different pages and I never saw any empty values